api_app/api/routes/workspace_service_templates.py

```diff
@@ -11,27 +11,27 @@
 from models.schemas.resource_template import ResourceTemplateInResponse, ResourceTemplateInformationInList
 from models.schemas.workspace_service_template import WorkspaceServiceTemplateInCreate, WorkspaceServiceTemplateInResponse
 from resources import strings
-from services.authentication import get_current_admin_user, get_current_tre_user
+from services.authentication import get_current_admin_user, get_current_workspace_owner_or_researcher_user_or_tre_admin
 
 
-service_templates_router = APIRouter(dependencies=[Depends(get_current_tre_user)])
-user_resource_templates_router = APIRouter(dependencies=[Depends(get_current_tre_user)])
+workspace_service_templates_router = APIRouter(dependencies=[Depends(get_current_workspace_owner_or_researcher_user_or_tre_admin)])
+user_resource_templates_router = APIRouter(dependencies=[Depends(get_current_workspace_owner_or_researcher_user_or_tre_admin)])
 
 
 # WORKSPACE SERVICE TEMPLATES
-@service_templates_router.get("/workspace-service-templates", response_model=ResourceTemplateInformationInList, name=strings.API_GET_WORKSPACE_SERVICE_TEMPLATES, dependencies=[Depends(get_current_tre_user)])
+@workspace_service_templates_router.get("/workspace-service-templates", response_model=ResourceTemplateInformationInList, name=strings.API_GET_WORKSPACE_SERVICE_TEMPLATES, dependencies=[Depends(get_current_workspace_owner_or_researcher_user_or_tre_admin)])
```

Comment:
get_current_tre_user became get_current_workspace_owner_or_researcher_user_or_tre_admin.
is that correct. isn't get_current_workspace_owner_or_researcher_user_or_tre_admin equals get_current_ws_user?

Reply:
same for rest of file

Reply:
`get_current_tre_user = AzureADAuthorization(require_one_of_roles=['TREUser'])`

whereas

`get_current_workspace_owner_or_researcher_user_or_tre_admin = AzureADAuthorization(require_one_of_roles=["TREAdmin", "WorkspaceOwner", "WorkspaceResearcher"])`

As some routes need to be accessed by both TREAdmins and any Workspace User. 

 get_current_workspace_owner_or_researcher_user_or_tre_admin does not equal get_current_ws_user as get_current_ws_user would not include TREAdmin. I renamed to be more explicit, so have:

 ```
 get_current_workspace_owner_user = AzureADAuthorization(require_one_of_roles=['WorkspaceOwner'])


get_current_workspace_researcher_user = AzureADAuthorization(require_one_of_roles=['WorkspaceResearcher'])


get_current_workspace_owner_or_researcher_user = AzureADAuthorization(require_one_of_roles=['WorkspaceOwner', 'WorkspaceResearcher'])
```

Let me know if want to talk through.

Reply:
talked offline. seems correct